### PR TITLE
[FEAT] New provisioner adapter to support HPCM

### PR DESCRIPTION
**Pull request by sindhu-106**
Signed-off-by: sindhuja <sindhuja.gajula@intel.com>

### CODE REVIEW

**Review by avincigu — DISMISSED** [review]

File: configurations/hw1/ProviderProvisionerNetworkForeignBus.json [hunk 1]
```diff
@@ -25,7 +23,7 @@
         "stateChangeSource"
       ],
       "subjects": [
-        "*"
+        "stateChanges"
```
Comment (paul-amonson) [inline configurations/hw1/ProviderProvisionerNetworkForeignBus.json:26]: Why add this line? The "*" covers everything.

File: networking/src/main/java/com/intel/networking/sink/kafka/NetworkDataSinkKafka.java [hunk 2]
```diff
@@ -53,7 +53,7 @@ public void initialize() {
             if(args_.containsValue(null) || args_.containsValue(""))
                 throw new NetworkException("Given argument value cannot be null or empty. argument: '" + requiredKafkaProperty + "'");
         }
-
+        parseSubjects(args_.getOrDefault(requiredKafkaProperties[3], "*"));
```
Comment (paul-amonson) [inline networking/src/main/java/com/intel/networking/sink/kafka/NetworkDataSinkKafka.java:56]: Consider using a constant here (and any other places). The "requiredKafkaProperties[3]" is obfuscated and requires the engineer to go find it. Also "3" becomes a "magic number". Making the code more readable is always a good thing. :)

File: configurations/hw1/ProviderProvisionerNetworkForeignBus.json [hunk 3]
```diff
@@ -6,16 +6,14 @@
...
       "arguments": {
-        "fullUrl": "https://api-gw-service-nmn.local/apis/sma-telemetry-api/v1/stream/cray-hmsstatechange-notifications?stream_id=dai-provisioner",
-        "inputBufferSize": "4",
-        "connectTimeout": "30",
-        "tokenAuthProvider": "com.intel.authentication.KeycloakTokenAuthentication",
-        "tokenServer": "https://api-gw-service-nmn.local/keycloak/",
-        "clientId": "admin-client",
-        "clientSecret": "dd0f8dee-f40b-4b16-a5dd-af2ed1b8a99c",
-        "realm": "shasta"
+        "bootstrap_servers": "admin:9092",
```
Comment (paul-amonson) [inline configurations/hw1/ProviderProvisionerNetworkForeignBus.json:9]: Is "admin" a hostname?

Comment (sindhu-106) [reply configurations/hw1/ProviderProvisionerNetworkForeignBus.json:9]: Yes Paul, these are the configuration details we got for kafka bus

File: configurations/hw1/ProviderProvisionerNetworkForeignBus.json (hunk 1 above)
Comment (sindhu-106) [reply configurations/hw1/ProviderProvisionerNetworkForeignBus.json:26]: To subscribe to kafka bus we do need to mention topic name.

File: configurations/hw1/ProviderProvisionerNetworkForeignBus.json [hunk 4]
```diff
@@ -6,16 +6,15 @@
...
-        "clientId": "admin-client",
-        "clientSecret": "dd0f8dee-f40b-4b16-a5dd-af2ed1b8a99c",
-        "realm": "shasta"
+        "bootstrap.servers": "admin:9092",
+        "group.id": "boot_states",
+        "schema.registry.url": "http://admin:8081",
+        "auto.commit.enable": false,
+        "auto.offset.reset": "earliest",
+        "specific.avro.reader": "true",
+        "topics": "stateChanges"
```
Comment (paul-amonson) [inline configurations/hw1/ProviderProvisionerNetworkForeignBus.json:15]: Is that actual kafka topic name "stateChanges"? Seems odd that its that same as our internal name for that data type. LOL

**Review by paul-amonson — APPROVED** [review]

**Review by paul-amonson — APPROVED** [review]

**Review by nbwhite — APPROVED** [review]

**Review by nsai1 — APPROVED** [review]

**Pull request merged**